src/main/java/com/sarahhand/fractals/colorpalette/ChangeColorPalette.java

```diff
@@ -0,0 +1,150 @@
+package com.sarahhand.fractals.colorpalette;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.JButton;
+import javax.swing.JColorChooser;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+
+import com.sarahhand.fractals.mapper.ColorPaletteMapper;
+import com.sarahhand.fractals.model.ColorPalette;
+
+/** This class changes the color palette to one of your choice using the class ColorPanel.
+ * 
+ * @author M00031
+ *
+ */
+public class ChangeColorPalette {
+
+	private JFrame frame;
+
+	private JPanel colorPanelsPanel;
+	private List<ColorPanel> colorPanels;
+
+	private volatile ColorPalette newColorPalette;
+
+	private JPanel topPanel;
+	private JButton changeEndColors;
+	private JButton resetColorPalette;
+
+	private JPanel bottomPanel;
+	private JButton addColor;
+	private JButton doneButton;
+
+	private final int FRAME_WIDTH = 500;
+	private final int FRAME_HEIGHT = 500;
+	private final int COLOR_PALETTE_LENGTH = 200;
```

Comment:
Initialize this to ColorPalette.COLOR_PALETTE_LENGTH, or delete this, and just use ColorPalette.COLOR_PALETTE_LENGTH everywhere instead.